DeviceConnectionAndReliabilityReadme.md

```diff
@@ -0,0 +1,62 @@
+# Azure IoT Device Client C# SDK
+
+## Device connection and messaging reliability
+
+### Overview
+
+In this document you will find information about
+
+- The connection authentication and renewal methods.
+- The reconnection logic and retry policies.
+- The timeout controls.
+
+### Connection authentication
+
+Authentication can be done using either
```

Comment:
even though TPM based devices eventually produce sas token, I think it is worth calling out those as a separate category  of authentication mechanism (TPM based auth is more secure and sas key based, and those are the ones customers are recommended to use in prod scenarios)

Reply:
Will add to the list